.pre-commit-config.yaml

```diff
@@ -1,30 +1,36 @@
-# yet another python formatter
-- repo: local
-  hooks:
-  # yapf = yet another python formatter
-  - id: yapf
-    name: yapf
-    entry: yapf
-    language: system
-    types: [python]
-    args: ["-i"]
-    exclude: &exclude_files >-
-      (?x)^(
-        ^docs/|
-      )$
-
-  # prospector: collection of linters
-  - id: prospector
-    language: system
-    types: [file, python]
-    name: prospector
-    description: "This hook runs Prospector: https://github.com/landscapeio/prospector"
-    entry: prospector
-    exclude: *exclude_files
-
-- repo: git://github.com/pre-commit/pre-commit-hooks
-  sha: v1.1.1
-  hooks:
-  - id: check-yaml
+repos:
```

Comment:
Would you be able to just outline the changes here regarding precommits, prospector and pylint?  I saw something on aiida-core about this but I don't quite understand what's changing here.

Reply:
What changes is that we are installing `yapf` directly from a revision so we don't have to install it as a package dependency anymore. We do the same for `pylint` except there we still need to install an actual version in the `extras_requires` since we need to run it as `[system]` because otherwise it won't see all other installed libraries and throw a bunch of false positive `import-errors`. Finally, we remove `prospector` which we were only using to run `pylint` but it is better to run it directly since `prospector` has caused a lot of dependency conflicts in the past

Reply:
It also allowed getting rid of the long `.pylintrc` file since we are only using a minimal set of those options. Ideally once all the tools support `pyproject.toml` we can get rid of all the individual config files for all these tools. I hate that there are a million config files in every repo top folder

Reply:
So why a revision?  Given that this is a tool (rather than lib), I would want the latest version, always.

Reply:
Any why not a package (extra) dependency?  This seemed reasonable to me.

Reply:
> So why a revision? Given that this is a tool (rather than lib),

Because here we need to be sure that each environment (both local, as remote on CI) use exactly the same version. Since tools like `yapf` and `pylint` can change their complaints in newer versions, it can happen that you run a different version locally compared to the CI, resulting in pre-commit running just fine locally, but failing horribly on the CI, which goes without saying, is rather frustrating. This way you also prevent that if you develop multiple packages in the same environment that might have slightly different versions, you don't get annoying warnings that wouldn't apply on the CI.